scripting/src/main/java/org/infinispan/scripting/impl/ScriptingManagerImpl.java

```diff
@@ -177,109 +175,41 @@ ScriptMetadata getScriptMetadata(String scriptName) {
-            return new NoOpFuture<T>(result);
+            return CompletableFuture.completedFuture(result);
          } else {
             ScriptEngine engine = getEngineForScript(metadata);
             T result = (T) engine.eval(getScriptCache().get(metadata.name()), bindings);
-            return new NoOpFuture<T>(result);
+            return CompletableFuture.completedFuture(result);
          }
       } catch (ScriptException e) {
          throw log.scriptExecutionError(e);
       }
    }
 
-   private ScriptMetadata extractMetadataFromScript(String name, String script) {
-      ScriptMetadata.Builder metadataBuilder = new ScriptMetadata.Builder();
-
-      metadataBuilder.property(MetadataProperties.NAME, name);
-      metadataBuilder.property(MetadataProperties.MODE, ExecutionMode.LOCAL.toString().toLowerCase());
-      int s = name.lastIndexOf(".") + 1;
-      if (s == 0 || s == name.length())
-         metadataBuilder.property(MetadataProperties.EXTENSION, DEFAULT_SCRIPT_EXTENSION);
-      else
-         metadataBuilder.property(MetadataProperties.EXTENSION, name.substring(s));
-      if (script.startsWith("//")) {
-         int state = KEY;
-         String key = null;
-         String value = null;
-         StringBuilder sb = new StringBuilder();
-         char ch = 0;
-         for (int pos = 2; ch != 10 && ch != 13 && pos < script.length(); pos++) {
-            ch = script.charAt(pos);
-            switch (state) {
-            case KEY:
-               switch (ch) {
-               case '=':
-                  key = sb.toString().toUpperCase();
-                  sb = new StringBuilder();
-                  state = VALUE;
-                  break;
-               case ' ':
-                  break;
-               default:
-                  sb.append(ch);
-               }
-               break;
-            case VALUE:
-               switch (ch) {
-               case ',':
-               case 10:
-               case 13:
-                  value = sb.toString();
-                  metadataBuilder.property(MetadataProperties.valueOf(key), value);
-                  sb = new StringBuilder();
-                  state = KEY;
-                  break;
-               case ' ':
-                  break;
-               default:
-                  sb.append(ch);
-               }
-               break;
-            }
-         }
-      }
-
-      return metadataBuilder.build();
-   }
-
    ScriptEngine getEngineForScript(ScriptMetadata metadata) {
-      String language = metadata.property(MetadataProperties.LANGUAGE);
-      if (language != null) {
-         if (scriptEnginesByLanguage.containsKey(language)) {
-            return scriptEnginesByLanguage.get(language);
-         } else {
-            ScriptEngine engine = scriptEngineManager.getEngineByName(language);
+      return metadata.language().map( (language) -> {
+         return scriptEnginesByLanguage.computeIfAbsent(language, (lang) -> {
```

Comment:
Just a matter of preference, but I like the way regular control look better when there's a lot of nesting. You could also move the `noEngineForScript` check at the end, and you'd end up with something like this:

      ScriptEngine engine;
      if (metadata.language().isPresent()) {
         engine = scriptEnginesByLanguage
               .computeIfAbsent(metadata.language().get(), scriptEngineManager::getEngineByName);
      } else {
         engine = scriptEnginesByExtension
               .computeIfAbsent(metadata.extension(), scriptEngineManager::getEngineByExtension);
      }
      if (engine == null) {
         throw log.noEngineForScript(metadata.name());
      } else {
         return engine;
      }